c++/src/Reader.cc

```diff
@@ -1133,7 +1133,7 @@ namespace orc {
     if (currentRowInStripe == 0) {
       startNextStripe();
     }
-    uint64_t rowsToRead =
+    uint64_t rowsToRead = currentStripe >= lastStripe ? 0 :
```

Comment:
Is it better to add a else branch to line 1094 like below:

```
if (currentStripe < lastStripe) {  // line 1094
  ...
} else {
  currentRowInStripe = rowsInCurrentStripe = 0;
  previousRow = firstRowOfStripe[lastStripe - 1] +
    footer->stripes(static_cast<int>(lastStripe - 1)).numberofrows();
}
```

Reply:
Sure. I think these are all better since they fix the issue inside `startNextStripe()` which is also used in `RowReaderImpl::seekToRow()`.

While adding test coverage for `seekToRow()`, I encountered another issue (ORC-1160) and I realized it also exists in 1.7.x releases. I think we need to resolve it first: #1100 . Then I'll rebase the current PR on it and continue to add tests.

Reply:
I continue the work after #1100 is merged. The current PR is rebased onto it.

I think all the above solutions fix the issue and don't have significant performance difference. So I tend to focus on readability. Added a new method `markEndOfFile()` to clean up some duplicated codes. Also added the test coverage on `seekToRow()`.